src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package com.dev.cinema.security;
+
+import com.dev.cinema.exceptions.AuthenticationException;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.UserService;
+import com.dev.cinema.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User userFromDB;
+        try {
+            userFromDB = userService.findByEmail(email);
```

Comment:
should we handle RuntimeException?

Reply:
Yeah, RuntimeException its mistake of programmer. I forgot

Reply:
Soo?)

Reply:
Now AuthenticationException is just exception when user writes wrong password and we can handle it? Or I didn't sleep enough because I don't understand)

Reply:
you pushed commit and delete this try catch, then another one and returned it back eventually)In the end we don't have any changes here and try catch block still here